src/codewhisperer/util/dependencyGraph/goDependencyGraph.ts

```
@@ -0,0 +1,236 @@
+            const content = await readFileAsString(filePath)
+            if (this.readPackageName(content) !== packageName) {
+                continue
+            }
+            fileList.push(filePath)
+        }
+
+        return fileList
+    }
+
+    override async isTestFile(content: string): Promise<boolean> {
+        const imports = this.readImports(content)
+        const filteredImports = imports.filter(importStr => {
+            return importStr.includes('"testing"')
+        })
+        return filteredImports.length > 0
+    }
+
+    override async generateTruncation(uri: Uri): Promise<Truncation> {
+        try {
+            const workspaceFolder = vscode.workspace.getWorkspaceFolder(uri)
+            if (workspaceFolder === undefined) {
+                this._pickedSourceFiles.add(uri.fsPath)
+            } else {
+                await this.searchDependency(uri)
+                await this.traverseDir(this.getProjectPath(uri))
+            }
+            await sleep(1000)
+            getLogger().verbose('Picked source files: [' + [...this._pickedSourceFiles].join(', ') + ']')
+            const truncDirPath = this.getTruncDirPath(uri)
+            this.copyFilesToTmpDir(this._pickedSourceFiles, truncDirPath)
+            const zipFilePath = this.zipDir(truncDirPath, CodeWhispererConstants.codeScanZipExt)
+            const zipFileSize = statSync(zipFilePath).size
+            return {
+                rootDir: truncDirPath,
+                zipFilePath,
+                scannedFiles: new Set(this._pickedSourceFiles),
+                srcPayloadSizeInBytes: this._totalSize,
+                zipFileSizeInBytes: zipFileSize,
+                buildPayloadSizeInBytes: 0,
+                lines: this._totalLines,
+            }
+        } catch (error) {
+            getLogger().error('Go dependency graph error caused by:', error)
+            throw new Error('Go context processing failed.')
```

Should we throw a toolkit error instead?
```suggestion
            throw ToolkitError.chain(error, 'Go context processing failed.')
```